### DISCUSSION TITLE
I need to optimize the project for seo, such as: generate corresponding html pages according to the data...

### DISCUSSION THREAD

**Discussion by tendecade** [nuxt/nuxt#22588] [Questions]
**Question 1:** 

If I want to dynamically add/edit data in the management background for a project after it goes online, the project that has already gone online will generate a corresponding file ending in .html and access it through .html. At the same time, customers with the prefix of .html can also Custom (the custom name is stored in this piece of data); I wonder what should I do in this case?

For example:
When I create a product in the background, the customer will fill in the corresponding url name (as a .html prefix), assuming it is 
 'super-user-friendly-air-fryer'; when visiting the official website, the route should look like this: '/shop/super-user-friendly-air- 
fryer.html'.

**Question 2:** 

And I have launched the project now, and the generated site map has only one home page. Am I missing something or is there something wrong with my packaging method? My packaging method is: npx nuxt build

**Question 3:** 

And my website has five languages, I want to know the search engine ranking for seo, should I distinguish between five domain names:
www.en.xxx.com / www.pl.xxx.com / www.es.xxx.com
for easy recording. Instead of all under one domain name, store the currently selected language type through cookies, and display content in different languages?

Thank you for your help. Thank you

**Comment by TheAlexLichter** [comment]
Q1: That is certainly possible (if needed). But instead you could also redirect all `*.html` calls to a prefix-free version.

Q2: Depends on your set up. Usually all static sites should be picked up, dynamic ones if they are linked somewhere.

Q3: IMO, Every domain should use their respective language (if it is language-based and not country-based). Don't forget to set hreflang properly
